staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -940,12 +941,20 @@ func (r *Request) newUnstructuredResponseError(body []byte, isTextResponse bool,
 	message := "unknown"
 	if isTextResponse {
 		message = strings.TrimSpace(string(body))
+	} else {
+		// parse the message in the body.
+		var status metav1.Status
+		err := json.Unmarshal(body, &status)
```

Comment:
this will not error, even if the body is not actually a status object (it would not error on `{"foo":"bar"}`, for example)